src/csg.h

```diff
@@ -0,0 +1,78 @@
+//
+//  csg.h
+//  robust_implicit_networks
+//
+//  Created by Yiwen Ju on 6/3/24.
+//
+
+#ifndef csg_h
+#define csg_h
+#include <fstream>
+#include <string>
+#include <nlohmann/json.hpp>
+
+
+#include "implicit_arrangement.h"
+
+///
+///The structure of a csg unit
+///A nested data structure where `operation` takes in a boolean operation, and the elements contain either a function index or later csg unit, where netaives represent the indices, and positives reprsent the index in the CSG tree.
+///e.g. {Intersection, -1, 2} represents an intersection operation between the first function and the second csg unit.
+///
+struct csg_unit{
+    int operation;
+    std::array<int, 2> elements;
+};
```

Comment:
I thought we decided to take a call back function instead of explicit CSG tree?

```c++
std::function<bool(std::vecrtor<bool>)> keep;
// The function keep takes a list of signs (one per implicit functions) and
// outputs a bool indicating whether the CSG results contains a point with 
// such sign combination.
```

This way, we can hide the specific implementation details used to evaluate the CSG tree.